well-formedness.md

```diff
@@ -320,6 +320,22 @@ All identifiers beginning with "iele." are reserved by the language and cannot b
 
 ```
 
+In order to correctly check names, we must convert escaped IELE names to their correct token representation.
+
+```k
+    syntax String ::= unescape(String) [function]
+                    | unescape(String, Int, StringBuffer) [function, klabel(unescapeAux)]
+ // -------------------------------------------------------------------------------------
+    rule unescape(S) => unescape(S, 1, .StringBuffer)
+    rule unescape(S, IDX, SB) => unescape(S, IDX +Int 1, SB +String substrString(S, IDX, IDX +Int 1))
+      requires IDX <Int lengthString(S) -Int 1 andBool substrString(S, IDX, IDX +Int 1) =/=K "\\"
+    rule unescape(S, IDX, SB) => unescape(S, IDX +Int 3, SB +String chrChar(String2Base(substrString(S, IDX +Int 1, IDX +Int 3), 16)))
+      requires IDX <Int lengthString(S) -Int 1 andBool substrString(S, IDX, IDX +Int 1) ==K "\\"
```

Comment:
I think the first part of `requires` here should be `IDX <Int lengthString(S) -Int 3`

Reply:
Never mind. It should be fine as is since the strings are guaranteed to be well-formed because they have been checked with the parser.

Reply:
We decided offline to leave it as is. The code is guaranteed to work correctly for well formed strings, which we are guaranteed to be operating on because the strings have already been checked by the parser